### ISSUE TITLE
ASCII encoding error when saving translations with Unicode characters

### ISSUE THREAD

**Issue by KJ7LNW** [pescheckit/python-gpt-po#69]
## Problem

When translating to languages with non-ASCII characters (e.g., German with umlauts), saving fails with:

```
'ascii' codec can't encode character '\xfc' in position 804: ordinal not in range(128)
```

## Root Cause

`polib.save()` uses the encoding from the PO file's metadata. If the `Content-Type` header in the .po file is missing or doesn't specify `charset=UTF-8`, polib defaults to ASCII.

## Reproduction

Translate any .po file to a language containing Unicode characters when the file's metadata lacks proper charset declaration:

```bash
gpt-po-translator --folder ./po --lang de --bulk
```

## Proposed Solution

Add a warning when loading PO files that are using ASCII encoding:

```python
def _load_po_file(self, po_file_path: str):
    po_file = polib.pofile(po_file_path)
    if po_file.encoding.upper() == 'ASCII':
        logging.warning(
            "PO file %s is using ASCII encoding. "
            "Add 'Content-Type: text/plain; charset=UTF-8' to the file header to support Unicode characters.",
            po_file_path
        )
    return po_file
```

This warns users to fix their PO file metadata properly rather than silently changing encoding.

## Alternative: Auto-fix

Alternatively, automatically set UTF-8 encoding after loading:

```python
po_file.encoding = 'UTF-8'
```

However, warning is preferable as it alerts users to fix the source metadata.
